### DISCUSSION TITLE
Communication between OpenDDS and RTI-Micro-DDS

### DISCUSSION THREAD

**Discussion by z79523** [OpenDDS/OpenDDS#4436] [Q&A] [answered]
Hi, 
I tried to build the communication between OpenDDS and RTI micro DDS, and I reused the publisher source code from DevGuideExamples/DCPS/Messenger/Publisher.cpp.  OpenDDS act as publisher, RTI Micro-DDS is subscriber. When configured the transport method as **rtps_udp**, the communication can be built and work well, however when I modified the transport method to **udp**, couldn't build the communication between these two entities.

Here is the ini file that work with OpenDDS publisher and RTI Micro DDS subscriber.
![image](https://github.com/OpenDDS/OpenDDS/assets/16408632/58a2e580-b310-472c-addb-dc17d598da9f)
Here is the ini file that can not work with OpenDDS publisher  and RTI Micro DDS subscriber.
![image](https://github.com/OpenDDS/OpenDDS/assets/16408632/c2c8dff6-6e11-42ca-8622-25f40bcd5b44)
Is there anyone who knows the reason?

Thanks!

**Comment by simpsont-oci** [comment, marked as answer]
Since you are trying to communicate between two different vendors (OpenDDS and RTI), you will need to use an interoperable transport protocol. Currently, the only interoperability specification for DDS is [DDSI-RTPS](https://www.omg.org/spec/DDSI-RTPS/2.5/About-DDSI-RTPS), which defines the RTPS/UDP transport (rtps_udp in OpenDDS). The other transports will only work between two OpenDDS participants.
